WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java

```diff
@@ -835,18 +848,26 @@ private void showAddMediaPopup() {
     private static final int ITEM_CHOOSE_PHOTO  = 2;
     private static final int ITEM_CHOOSE_VIDEO  = 3;
 
+    private int getAddMenuItemFromPosition(int position) {
```

Comment:
I have weird feelings about this function. Because of the change in position of the menu items (based on which mode we're in) this function tries to return the correct position/item. I think this might be an indication that the positioning logic is spread across too many places in the code.

Since we are in the process of touching this part of the code, may I suggest refactoring it a bit so to deal with the mode-related menu logic in a single place? One way to do this is by using a [SimpleAdapter](https://developer.android.com/reference/android/widget/SimpleAdapter.html) instead of the `ArrayAdapter` to pass the information of which item got clicked. [Here's a gist](https://gist.github.com/hypest/9ba45d3b60f18e56e02bd84d5cc6e1d8) I put together with this idea. The menu items are [chosen in a single place](https://gist.github.com/hypest/9ba45d3b60f18e56e02bd84d5cc6e1d8#file-mediabrowseractivity-java-L813-L821) and then, the position of the clicked item plays no role since we're only interested in [the item clicked](https://gist.github.com/hypest/9ba45d3b60f18e56e02bd84d5cc6e1d8#file-mediabrowseractivity-java-L831). Let me know if you would be game to try that.

Reply:
I confess to not being truly happy with that routine, but I think resorting to a `SimpleAdapter` is a more complicated solution for what is really a minor problem. If the issue is that the logic appears spread out, I could always simply move that logic out of a separate method and have it within `doAddMediaItemClicked()` since that's the only place that calls it.

Reply:
I should have explained what I mean with "spread out" a bit more:
* There's code that populates an array with some strings ([Example](https://github.com/wordpress-mobile/WordPress-Android/blob/eb0773770d7407baf742933533efe918ac3ceb0d/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java#L787)) the order of which is important as it's implied in the onClick handler.
* The order of the strings depends on a mode flag (mBrowserType). That mode flag is then needed in the handler to interpret the clicked position.
* There are constants defined ([here](https://github.com/wordpress-mobile/WordPress-Android/blob/eb0773770d7407baf742933533efe918ac3ceb0d/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java#L847-L850)) but only used in the onClick handler, not when the order is defined. Those two can get out of sync if we make a change in the code. They also get out of sync in runtime because of the point mentioned next:
* The name of those constants imply an action but the integer they are based on is just a position. The position varies due to the mode and that's what `getAddMenuItemFromPosition` tries to fix up.

All in all, the above seems like a rather complex and error prone way of just putting up a popup and handling its onClick.

I'm not insisting on the specific solution in the gist but, notice how the gist is free from any logic that depends on the order or position, decoupling the onClick handler from such detail.

In the end, the code in this PR works OK anyway and a refactoring to make it more readable/less error prone can be considered an optimisation so, let me know and I can merge it as is.

Reply:
Hmm, I see your point. I took your advice and implemented a `SimpleAdapter` in 4a7a539.

Reply:
Thanks for the change Nick!

Took the liberty to touch a couple of lines for a quick code style fix with 1054517.